core/chains/evm/logpoller/log_poller.go

```diff
@@ -612,12 +617,82 @@ func (lp *logPoller) LatestLogEventSigsAddrsWithConfs(fromBlock int64, eventSigs
 
 // GetBlocks tries to get the specified block numbers from the log pollers
 // blocks table. Returns the blocks it was able to find, empty slice if none.
-// It is not guaranteed to find the blocks specified, since
-// the log poller does not save blocks when backfilling.
-// It can be used as a cache of recent blocks (falling back to an RPC on cache miss),
-// to speed up querying and avoid loading an EVM node.
-func (lp *logPoller) GetBlocks(numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
-	return lp.orm.GetBlocks(numbers, qopts...)
+// When the log poller does not have requested blocks, it falls back
+// to RPC to fetch the missing blocks.
+// response contains blocks in the same order as "numbers" in request parameters
+func (lp *logPoller) GetBlocks(ctx context.Context, numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
+	blocksFound := make(map[uint64]LogPollerBlock)
+	qopts = append(qopts, pg.WithParentCtx(ctx))
```

Comment:
This is usually provided by callers, does this `WithParentCtx` override previously provided `WithParentCtx` options?

Reply:
it would, I think thats ok here

Reply:
yeah the `ctx` will override the contexts provided in `qopts`. We iterate through `qopts` array [here](https://github.com/smartcontractkit/chainlink/blob/develop/core/services/pg/q.go#L123-L125) 

Adding a comment for this behavior.

Reply:
ah didn't see connor commented already